src/app/core/entity-components/entity-details/form/form.component.ts

```diff
@@ -128,8 +128,10 @@ export class FormComponent implements OnInitDynamicComponent, OnInit {
   private assignFormValuesToEntity(entity: Entity, form: FormGroup) {
     Object.keys(form.controls).forEach((key) => {
       const value = form.get(key).value;
-      if (value !== null) {
+      if (value || value === 0) {
```

Comment:
checkboxes setting a value `false` should also be allowed. This check seems like something that will make us stumble in the future again.

Maybe we should allow any value to be set (at least if the form field has been changed) and let the schema datatypes handle special requirements (e.g. if null should not be written to the database)?

Reply:
btw.: good investigation of the problem, @RichardNutt !

Reply:
@sleidig I agree that this check seems like a bad idea. I just realized that we are already filtering undefined entries
https://github.com/Aam-Digital/ndb-core/blob/80538b02af7f8bd8ba0b072b98088e2f3bb82dcf/src/app/core/entity/schema/entity-schema.service.ts#L165-L172

Checking for null etc. in `ndb-core/src/app/core/entity/schema-datatypes/` seems better (and perhaps print some info to the console if values are omitted to make debugging easier). Should we just remove null/empty string/etc to keep the DB smaller?

Reply:
For now, there is not really a reason to distinguish between `null` and `undefined` in the database and I thing also empty strings or arrays. I can just imagine that this might interfere with the default values set in the classes when creating a new entity, e.g. in the child entity `status` is set to `""` on default. When the database now has `status === undefined` or `status === null` the `Object.assign` call in the schema transformation will overwrite the default value. If however the property is not present in the database, the value will not be overwritten. 

So I think it would make sense to delete all _empty_ values (`null`, `undefined`, `[]`, `""`) before writing a entity to the database, but make sure to not throw away zeros or `false` values.